lblod/frontend-reglementaire-bijlage
# Enable editing of imported resources when editing snippets

## Overview
Move to editor and plugins that support rdfa linking of imported resources in snippets. Biggest change here is to update the list of imported resources on the snippet list whenever saving a snippet.

Known issues:
- [x] Deleting a relationship on a document node (so an imported resource relationship) doesn't work, it's necessary to delete the backlink and reload
- [ ] Relationship editor doesn't display the subject for imported resource relationships, which is confusing
- [ ] There's no good handling of unlinked imported resources, e.g. preventing insertion if not linked or just removing the backlinks
- [x] Deleting snippets from a snippet list does not update the list of imported resources for that list (e.g. if an import is no longer used by any snippets, it should be removed but is not).

##### connected issues and PRs:
Plugins: https://github.com/lblod/ember-rdfa-editor-lblod-plugins/pull/462
Editor: https://github.com/lblod/ember-rdfa-editor/pull/1212
backend: https://github.com/lblod/app-reglementaire-bijlage/pull/81
For Jira ticket [GN-4898](https://binnenland.atlassian.net/browse/GN-4898)
GN: https://github.com/lblod/frontend-gelinkt-notuleren/pull/704

### Setup
Needs to be run with a backend that supports storing imported resources on snippet lists.

### How to test/reproduce
There are 3 steps to this:
1. Create a snippet with imported resources:
	- Go to the snippets page
	- Create a new or edit an existing snippet list, then edit or create a snippet
	- Create arbitrary RDFa in the editor
	- Click on the 'document' node (so click on the page outside of the content), and verify that the RDFa tools give the type 'document'
	- Open the 'Add Relationship' modal, create a link as normal, except also specify the subject. For the first imported resource you'll need to enter the placeholder URI into the power-select 'search' and hit enter, this will set the field to this new value. Later you can select an existing import or create a new one in the same way.
	- Hit save
	- Go back to 'edit snippet list' or 'manage snippet lists' pages and verify that the snippet lists the union of all the imported resources used in each of the snippets in that list
2. Create a template containing a snippet placeholder:
	- Go to the template management page and create or edit a template
	- Create some arbitrary RDFa, including at least one resource
	- Insert a snippet placeholder which incudes imported resources
	- Select the placeholder and see that the RDFa relationship editor has a new section listing the imported resources. You can click on the `+` to choose the 'real' resource URI that it links to in the document. Clicking the name of the linked 'real' resource allows you to change an already configured link.
	- Click save and publish, then publish.
3. Create a real document from a template with a snippet placeholder:
	- See [PR in GN](https://github.com/lblod/frontend-gelinkt-notuleren/pull/704)

### Challenges/uncertainties
N/A

### Checks PR readiness
